web/src/lib/utils/auth.ts

```diff
@@ -19,19 +21,19 @@ export const getAuthUser = async () => {
 
 export const authenticate = async (options?: AuthOptions) => {
   options = options || {};
+  const isAuthenticated = getAuthCookie() === 'true';
 
   const savedUser = getSavedUser();
-  const user = savedUser || (await getAuthUser());
-
-  if (!user) {
-    redirect(302, AppRoute.AUTH_LOGIN);
-  }
-
-  if (options.admin && !user.isAdmin) {
-    redirect(302, AppRoute.PHOTOS);
+  const user = savedUser || isAuthenticated ? await getAuthUser() : null;
+  if (!options.public) {
+    if (!user) {
+      redirect(302, AppRoute.AUTH_LOGIN);
+    }
+    if (options.admin && !user.isAdmin) {
+      redirect(302, AppRoute.PHOTOS);
+    }
   }
-
-  if (!savedUser) {
+  if (!savedUser && user) {
```

Comment:
this is quite messy. can you remove nested if statements and clean up the logic so it is more readable.